src/System.Text.Json/src/System/Text/Json/Serialization/ReflectionEmitMemberAccessor.cs

```diff
@@ -0,0 +1,152 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#if BUILDING_INBOX_LIBRARY
+using System.Diagnostics;
+using System.Reflection;
+using System.Reflection.Emit;
+using System.Text.Json.Serialization.Converters;
+
+namespace System.Text.Json
+{
+    internal sealed class ReflectionEmitMemberAccessor : MemberAccessor
+    {
+        public override JsonClassInfo.ConstructorDelegate CreateConstructor(Type type)
+        {
+            Debug.Assert(type != null);
+            ConstructorInfo realMethod = type.GetConstructor(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance, binder: null, Type.EmptyTypes, modifiers: null);
+
+            if (realMethod == null && !type.IsValueType)
+            {
+                return null;
+            }
+
+            var dynamicMethod = new DynamicMethod(
+                ConstructorInfo.ConstructorName,
+                typeof(object),
+                Type.EmptyTypes,
+                typeof(ReflectionEmitMemberAccessor).Module,
+                skipVisibility: true);
+
+            ILGenerator generator = dynamicMethod.GetILGenerator();
+
+            if (realMethod == null)
+            {
+                LocalBuilder local = generator.DeclareLocal(type);
+
+                generator.Emit(OpCodes.Ldloca_S, local);
+                generator.Emit(OpCodes.Initobj, type);
+                generator.Emit(OpCodes.Ldloc, local);
+                generator.Emit(OpCodes.Box, type);
```

Comment:
We should preserve the current functionality where we don't box\unbox. Both validators and get\set were previously there to prevent boxing for performance. We need to run the existing serialization benchmarks.

Reply:
This code wasn't changed.

Reply:
Sorry comment was meant for getter\setter below.